ttnn/cpp/ttnn/operations/data_movement/tilize_with_val_padding/device/tilize_with_val_padding_op.cpp

```diff
@@ -82,7 +82,12 @@ operation::ProgramWithCallbacks TilizeWithValPadding::create_program(
     if (input_tensor_a.memory_config().is_sharded() || this->use_multicore) {
         return detail::tilize_with_val_padding_multi_core(input_tensor_a, output_tensor, this->pad_value);
     }
-    return detail::tilize_with_val_padding_single_core(input_tensor_a, output_tensor, this->pad_value);
+    if(std::holds_alternative<int>(this->pad_value)) {
```

Comment:
what's the distinction between these two branches? The call site seems the same. Is the difference between float and int here?